### PR TITLE
Expose party's profile in parties API

### LINKED ISSUE #10375: Expose party's profile API
The parties API should expose the profile information.

### PR DESCRIPTION
**Pull request by ValentinTrinque**
Close #10375

### CODE REVIEW

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 1]
```diff
@@ -2428,10 +2428,21 @@ message ListPartiesResponse {
   PartyConnection parties = 1;
 }
 
+// Party represents an entity that is trading on the VEGA network.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:2431]:
```suggestion
// Party represents an entity that is using the VEGA network.
```
You can have a party ID and never trade :)

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 2]
```diff
@@ -2411,7 +2411,7 @@ message GetPartyRequest {
 // Response from getting party by ID
 message GetPartyResponse {
   // Data for the party requested.
-  vega.Party party = 1;
+  Party party = 1;
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:2414]: Is it ok to delete this and not have a breaking change?

File: datanode/sqlsubscribers/party.go [hunk 3]
```diff
@@ -30,8 +31,14 @@ type PartyEvent interface {
 	Party() types.Party
 }
 
+type PartyProfileUpdatedEvent interface {
+	events.Event
+	PartyProfileUpdated() eventspb.PartyProfileUpdated
```
Comment (jeremyletang) [inline datanode/sqlsubscribers/party.go:36]: Core method have the following signature
```suggestion
	PartyProfileUpdated() *eventspb.PartyProfileUpdated
```

File: protos/sources/data-node/api/v2/trading_data.proto (hunk 2 above)
Comment (ValentinTrinque) [reply protos/sources/data-node/api/v2/trading_data.proto:2414]: Actually, we are not going down that route. will revert it.

File: datanode/gateway/graphql/schema.graphql [hunk 4]
```diff
@@ -942,6 +950,14 @@ type Query {
     pagination: Pagination
   ): PartyConnection
 
+  "List all parties profiles by their IDs. If no id is set, all profiles are returned."
+  partiesProfilesConnection(
+    "Restrict the returned parties profiles to only the given party IDs. If not set, all parties profiles will be returned."
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:955]:
```suggestion
  "List parties' profiles by their IDs. If no ID is set, all profiles are returned."
  partiesProfilesConnection(
    "Restrict the returned parties' profiles to only the given party IDs. If not set, all profiles will be returned."
```

File: datanode/gateway/graphql/schema.graphql [hunk 5]
```diff
@@ -5656,6 +5672,30 @@ type PartyConnection {
...
+type PartyProfileEdge {
+  "The party's profile"
+  node: PartyProfile!
+  "The cursor for this party's profile"
+  cursor: String!
+}
+
+"Connection type for retrieving cursor-based paginated parties profiles information"
+type PartiesProfilesConnection {
+  "The parties profiles in this connection"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:5693]:
```suggestion
"Connection type for retrieving cursor-based paginated profile information for multiple parties"
type PartiesProfilesConnection {
  "The parties profiles in this connection"
```

Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:5693]:
```suggestion
"Connection type for retrieving cursor-based paginated parties profiles information"
type PartiesProfilesConnection {
  "The profiles in this connection"
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 6]
```diff
@@ -395,6 +395,13 @@ service TradingDataService {
     option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {tags: "Parties"};
   }
 
+  // List parties profiles
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:398]:
```suggestion
  // List parties' profiles
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 7]
```diff
@@ -395,6 +395,13 @@ service TradingDataService {
...
+  //
+  // Get a list of parties profiles
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:400]:
```suggestion
  // Get a list of profiles for multiple parties
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 8]
```diff
@@ -2444,6 +2451,37 @@ message PartyConnection {
   PageInfo page_info = 2;
 }
 
+// Request to list parties profiles by party ID.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:2454]:
```suggestion
// Request to list profiles by party ID.
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 9]
```diff
@@ -2444,6 +2451,37 @@ message PartyConnection {
...
+message ListPartiesProfilesRequest {
+  // Restrict the returned parties profiles to only the given party IDs. If not set,
+  // all parties profiles will be returned.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:2457]:
```suggestion
  // Restrict the returned profiles to only the given party IDs. If not set,
  // all parties' profiles will be returned.
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 10]
```diff
@@ -2444,6 +2451,37 @@ message PartyConnection {
...
+  repeated string parties = 1;
+  // Pagination control.
+  Pagination pagination = 2;
+}
+
+// Response from listing parties profiles.
+message ListPartiesProfilesResponse {
+  // Page of parties profiles data and corresponding page information.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:2465]:
```suggestion
// Response from listing parties' profiles.
message ListPartiesProfilesResponse {
  // Page of profile data and corresponding page information.
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 11]
```diff
@@ -2444,6 +2451,37 @@ message PartyConnection {
...
+
+// Party's profile data with the corresponding cursor.
+message PartyProfileEdge {
+  // Data associated with a party's profile.
+  vega.PartyProfile node = 1;
+  // Cursor that can be used to fetch further pages.
+  string cursor = 2;
+}
+
+// Page of parties profiles data and corresponding page information.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:2477]:
```suggestion
// Page of profile data per party and corresponding page information.
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 12]
```diff
@@ -2444,6 +2451,37 @@ message PartyConnection {
...
+message PartiesProfilesConnection {
+  // Page of parties profiles and their corresponding cursors.
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:2479]:
```suggestion
  // Page of profiles and their corresponding cursors.
```

**Review by EVODelavega — DISMISSED** [review]

**Review by candida-d — DISMISSED** [review]

**Review by EVODelavega — APPROVED** [review]

**Review by candida-d — APPROVED** [review]

**Pull request merged**